symbology/read-symbols-from-mobile-style-file/src/main/java/com/esri/samples/read_symbols_from_mobile_style_file/ReadSymbolsFromMobileStyleFileController.java

```diff
@@ -0,0 +1,356 @@
+
+    // get the symbol from the SymbolStyle
+    ListenableFuture<Symbol> symbolFuture = emojiStyle.getSymbolAsync(symbolKeys);
+    symbolFuture.addDoneListener(() -> {
+      try {
+        faceSymbol = (MultilayerPointSymbol) symbolFuture.get();
+        if (faceSymbol == null) {
+          return;
+        }
+
+        // loop through all the symbol layers and lock the color
+        faceSymbol.getSymbolLayers().forEach(symbolLayer -> symbolLayer.setColorLocked(true));
+
+        // unlock the color of the base layer. Changing the symbol layer color will now only change this layer's color
+        faceSymbol.getSymbolLayers().get(0).setColorLocked(false);
+
+        // set the color of the symbol
+        faceSymbol.setColor(colorSelectionListView.getSelectionModel().getSelectedItem());
+
+        // set the size of the symbol
+        faceSymbol.setSize((float) sizeSlider.getValue());
+
+        // update the symbol preview
+        updateSymbolPreview(faceSymbol);
+
+      } catch (ExecutionException | InterruptedException e) {
+        new Alert(Alert.AlertType.ERROR, "Error creating symbol with the provided symbol keys" + e.getMessage()).show();
+      }
+    });
+  }
+
+  /**
+   * Returns the symbol key of the symbol style search result that is currently selected in the list view.
+   *
+   * @param listView the list view of which to query the selected item
+   * @return String representing the key of the selected symbol style
+   */
+  private String getRequestedSymbolKey(ListView<SymbolStyleSearchResult> listView) {
+    SymbolStyleSearchResult requestedSymbol = listView.getSelectionModel().getSelectedItem();
+    return (requestedSymbol != null ? requestedSymbol.getKey() : "");
+  }
+
+  /**
+   * Creates an ImageView object from a provided symbol and displays it in the preview area.
+   *
+   * @param multilayerPointSymbol the symbol used to create the image view
+   */
+  private void updateSymbolPreview(MultilayerPointSymbol multilayerPointSymbol) {
+    // remove the previously displayed image view
+    symbolPreviewHBox.getChildren().clear();
+
+    // create an image and image view from the symbol
+    ListenableFuture<Image> symbolImageFuture = multilayerPointSymbol.createSwatchAsync(0x00000000, 1);
+    try {
+      Image symbolImage = symbolImageFuture.get();
+      ImageView symbolImageView = new ImageView(symbolImage);
+      // display the image view in the preview area
+      symbolPreviewHBox.getChildren().add(symbolImageView);
+
+    } catch (InterruptedException | ExecutionException e) {
+      new Alert(Alert.AlertType.ERROR, "Error creating preview ImageView from provided MultilayerPointSymbol" + e.getMessage()).show();
+    }
+  }
+
+  /**
+   * Updates the size of the symbol preview and the label when the slider is interacted with.
+   */
+  @FXML
+  private void updateSymbolSize() {
+    // get the slider value and convert to a string
+    String sizeValue = Integer.toString((int) sizeSlider.getValue());
```

Comment:
Can you just add ticks to the slider?

Reply:
Can do. Removed that method in the process.